diff --git a/src/Plugin/conditional_fields/handler/Select.php b/src/Plugin/conditional_fields/handler/Select.php
index b3cdfd9..663abcb 100644
--- a/src/Plugin/conditional_fields/handler/Select.php
+++ b/src/Plugin/conditional_fields/handler/Select.php
@@ -44,10 +44,10 @@ class Select extends ConditionalFieldsHandlerBase {
         $regex = TRUE;
       case CONDITIONAL_FIELDS_DEPENDENCY_VALUES_NOT:
       case CONDITIONAL_FIELDS_DEPENDENCY_VALUES_OR:
-        foreach ($options['values'] as $value) {
+        foreach ((array) $options['values'] as $value) {
           $select_states[$options['state']][] = [
             $options['selector'] => [
-              $options['condition'] => empty($regex) ? [$value] : $options['value'],
+              $options['condition'] => empty($regex) ? $value : $options['value'],
             ],
           ];
         }
@@ -76,10 +76,10 @@ class Select extends ConditionalFieldsHandlerBase {
     }
     else {
       $values = array_column($options['value_form'], $key_column);
-      $state[$options['state']][$options['selector']] = array('value' => $values);
+      $state[$options['state']][$options['selector']] = ['value' => $values];
     }

     return $state;
   }

-}
\ No newline at end of file
+}
